docs/vgclog_examples.md

```diff
@@ -0,0 +1,594 @@
+
+<!--A blank line appears in the log after this `<exclusive-end>` line and before the next section, which tells you that no STW activities are running. However, concurrent activities might be running. In this case, the concurrent operations  and sub-operations of the second increment of the `gencon` global cycle are running.-->
+
+The next section of the logs records an STW pause that is associated with an allocation failure. The `<cycle-start>` element that follows this STW pause indicates that the cycle is a `scavenge` cycle, which is the partial GC cycle used by the `gencon` GC:
+
+```xml
+...
+<cycle-start id="12368" type="scavenge" contextid="0" timestamp="2020-10-18T13:35:44.582" intervalms="580.047" />
+...
+```
+
+Subsequent elements have a `contextid=“12368”` which matches the `id` of this new `scavenge`cycle. For detailed information about how this cycle is recorded in the logs, see [Scavenge partial GC cycle](#scavenge-partial-gc-cycle).
+
+
+<!---
+```xml
+<exclusive-start id="12366" timestamp="2020-10-18T13:35:44.582" intervalms="237.874">
+
+<response-info timems="0.094" idlems="0.033" threads="5" lastid="00000000014E0F00" lastname="LargeThreadPool-thread-67" />
+
+</exclusive-start>
+
+...
+
+<af-start id="12367" threadId="00000000013D7280" totalBytesRequested="96" timestamp="2020-10-18T13:35:44.582" intervalms="580.045" type="nursery" />
+
+<cycle-start id="12368" type="scavenge" contextid="0" timestamp="2020-10-18T13:35:44.582" intervalms="580.047" />
+
+...
+
+<exclusive-end id="12377" timestamp="2020-10-18T13:35:44.594" durationms="11.816" />
+```
+--->
+
+The operations and sub-operations of the second, concurrent increment of the `gencon` global cycle are paused while the STW `scavenge` operation is running, and resume when the STW pause finishes. 
+
+After the partial GC cycle completes and the STW pause finishes, the log records a new STW pause, which is triggered to enable the final `gencon` global GC increment to run. This final increment finishes marking the nursery area and completes the global cycle. The `<exclusive-start>` element is followed by a `<concurrent-global-final>` element which logs the beginning of this final increment (and by implication, the end of the second increment).
+
+```xml
+<exclusive-start id="12378" timestamp="2020-10-18T13:35:44.594" intervalms="12.075">
+
+<response-info timems="0.108" idlems="0.040" threads="3" lastid="00000000018D3800" lastname="LargeThreadPool-thread-33" />
+
+</exclusive-start>
+
+<concurrent-global-final id="12379" timestamp="2020-10-18T13:35:44.594" intervalms="516905.029" >
+
+<concurrent-trace-info reason="card cleaning threshold reached" tracedByMutators="200087048" tracedByHelpers="12164180" cardsCleaned="4966" workStackOverflowCount="0" />
+
+</concurrent-global-final>
+```
+
+The final increment of a global cycle is triggered when the card-cleaning threshold is reached, indicated by the `reason` attribute of the `<concurrent-trace-info>` element.
```

Comment:
As previosuly mentioned, the could be 'tracing completed' reason too.

And actually there are more corner scenarios how we can trigger the final increment, such as percolate global. But that would warrant a whole other section that we may eventually cover (I need to provide examples first). That would be advanced, but important (for debugging) topic to cover.

Reply:
The `reason` attribute of the `<concurrent-trace-info>` child element indicates that this final STW increment of the global cycle has been triggered because a card-cleaning threshold has been reached. The concurrent tracing has been terminated prematurely and the `targetBytes` concurrent tracing target, recorded at the cycle start by `<concurrent-kickoff>`,  has not been achieved concurrently. If the concurrent tracing completes without interruption, the `<concurrent-trace-info` element logs `reason=tracing completed`.